### ISSUE TITLE
Workflow PublishItemWorkflowAction uses PublishSingle

### ISSUE THREAD

**Issue by jules0x** [symbiote/silverstripe-advancedworkflow#378]
The `execute` function in PublishItemWorkflowAction.php uses publishSingle() instead of publishRecursive() which means $owns relationships are not published.
This was noticed specifically on Elemental as edits to components should be published when the owning page is published.

**Comment by robbieaverill** [comment]
Hi @jules0x, I've had a look through the history of this file and my guess would be that at the time, to recommended replacement for `doPublish()` was `publishSingle()`. I'd be happy for you to make a PR to change it to `publishRecursive()` (which is now the recommended replacement for `doPublish()`).

Would you like to make a PR?

**Comment by jules0x** [comment]
Sure, I'll do that some time today.

**Comment by robbieaverill** [comment]
Thanks, please make the PR against the 5.0 branch (I'll merge it up afterwards)

**Comment by jules0x** [comment]
🤔 I created this branch off **master** so my PR has 27 commits in it, though I only made the 1. Should I have branched off 5.0 to make the adjustment instead?

**Comment by ScopeyNZ** [comment]
Yeah. You can just perform a `git rebase origin/5.0` and force-push your branch to fix this. I can do this if you want?

**Comment by robbieaverill** [comment]
I've rebased the PR
